crypto/encoding/proto/types.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+package tendermint.crypto.encoding.proto;
+
+option go_package = "github.com/tendermint/tendermint/crypto/encoding/proto";
+
+import "third_party/proto/gogoproto/gogo.proto";
+import "third_party/proto/cosmos-proto/cosmos.proto";
+
+option (gogoproto.populate_all) = true;
+option (gogoproto.equal_all)    = true;
+option (gogoproto.testgen_all)  = true;
+
+message PubKey {
+  option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PubKey";
+  oneof key {
+    bytes ed25519 = 1
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/ed25519.PubKey"];
+    bytes secp256k1 = 2
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/secp256k1.PubKey"];
+    bytes sr25519 = 3
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/sr25519.PubKey"];
+    // TODO: implement
+    // PubKeyMultiSigThreshold multisig = 4;
+    // [(gogoproto.casttype) =
+    //      "github.com/tendermint/tendermint/crypto/multisig.PubKeyMultisigThreshold"];
+  }
+}
+
+message PrivKey {
+  option (cosmos_proto.interface_type) = "*github.com/tendermint/tendermint/crypto.PrivKey";
+  oneof key {
+    bytes ed25519 = 1
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/ed25519.PrivKey"];
+    bytes secp256k1 = 2
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/secp256k1.PrivKey"];
+    bytes sr25519 = 3
+        [(gogoproto.casttype) = "github.com/tendermint/tendermint/crypto/sr25519.PrivKey"];
+  }
```

Comment:
I'm wondering why we'd ever need to send this (any private key material) over the wire. Having a proto message for this seems wrong to me. If it's kept here it should come with a noisy warning, why it's here, how it's used and that you should never send it over the wire.